lib/puppet/type/puppet_agent_abort_run.rb

```diff
@@ -0,0 +1,30 @@
+require 'puppet/property/boolean'
+
+Puppet::Type.newtype(:puppet_agent_abort_run) do
```

Comment:
`Abort` makes me think there was an error during the run. Maybe refer to the type and provider as `:puppet_agent_restart`?

Reply:
`restart` would imply that the run is restarted, that only happens when running as a daemon however.

I'll try to find a better name.

Reply:
`puppet_agent_end_run`